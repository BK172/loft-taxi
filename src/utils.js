import { NameSpace } from './store/reducers/root-reducer';

export const extend = (a, b) => Object.assign({}, a, b);

export const drawRoute = (map, coordinates) => {
  map.flyTo({
    center: coordinates[0],
    zoom: 15,
  });

  map.addLayer({
    id: 'route',
    type: 'line',
    source: {
      type: 'geojson',
      data: {
        type: 'Feature',
        properties: {},
        geometry: {
          type: 'LineString',
          coordinates,
        }
      }
    },
    layout: {
      'line-join': 'round',
      'line-cap': 'round',
    },
    paint: {
      'line-color': '#ffc617',
      'line-width': 8,
    }
  });
};

export const makeInitialStateMock = () => ({
  [NameSpace.AUTH]: {
    isLoggedIn: false,
    authToken: 'AUTH_TOKEN',
  },
  [NameSpace.CARD]: {
    cvc: '123',
    cardName: 'CardName',
    cardNumber: '111111111111',
    expiryDate: '01/11',
  },
});