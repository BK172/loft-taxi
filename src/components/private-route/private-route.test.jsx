import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import { makeInitialStateMock } from '../../utils';
import PrivateRoute from './private-route';

describe(`PrivateRoute component test`, () => {
  const store = configureStore()(makeInitialStateMock());

  it(`Should render PrivateRoute correctly if isLoggedIn true`, () => {
    const tree = renderer
      .create(
          <Provider store={store}>
            <MemoryRouter>
              <PrivateRoute
                component={() => <div>Component content</div>}
                isLoggedIn={true}
                exact={true}
                path={``}
              />
            </MemoryRouter>
          </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it(`Should render PrivateRoute correctly if isLoggedIn false`, () => {
    const tree = renderer
      .create(
          <Provider store={store}>
            <MemoryRouter>
              <PrivateRoute
                component={() => <div>Component content</div>}
                isLoggedIn={false}
                exact={true}
                path={``}
              />
            </MemoryRouter>
          </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});