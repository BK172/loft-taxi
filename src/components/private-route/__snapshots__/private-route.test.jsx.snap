// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrivateRoute component test Should render PrivateRoute correctly if isLoggedIn false 1`] = `null`;

exports[`PrivateRoute component test Should render PrivateRoute correctly if isLoggedIn true 1`] = `null`;
